internal/castai/castai.go

```diff
@@ -58,9 +50,10 @@ type Client interface {
 }
 
 // NewClient creates and configures the CAST AI client.
-func NewClient(log *logrus.Logger, rest *resty.Client, deltaHTTPClient *http.Client) Client {
+func NewClient(log logrus.FieldLogger, localLog logrus.FieldLogger, rest *resty.Client, deltaHTTPClient *http.Client) Client {
```

Comment:
The responsibilities have gotten mixed up. The castai client is responsible for sending requests to the CAST AI API, while the logs exporter is responsible for exporting the logs. Shouldn't the logic that is responsible for breaking the circuit of sending error logs about failed export be in the logs exporter package?